server/src/main/java/org/candlepin/controller/CandlepinPoolManager.java

```diff
@@ -1630,102 +1637,35 @@ public Entitlement adjustEntitlementQuantity(Consumer consumer,
         Map<String, Integer> poolQuantityMap, CallerType caller)
         throws EntitlementRefusedException {
 
-        // Because there are several paths to this one place where entitlements
-        // are granted, we cannot be positive the caller obtained a lock on the
-        // pool when it was read. As such we're going to reload it with a lock
-        // before starting this process.
-        log.debug("Locking pools: {}", poolQuantityMap.keySet());
-        List<Pool> pools = poolCurator.lockAndLoadBatchById(poolQuantityMap.keySet());
-        if (log.isDebugEnabled()) {
-            for (Pool pool : pools) {
-                log.debug("Locked pool: {} consumed: {}", pool, pool.getConsumed());
-            }
-        }
-
-        Map<String, PoolQuantity> poolQuantities = new HashMap<String, PoolQuantity>();
-        boolean quantityFound = false;
-        for (Pool pool : pools) {
-            Integer quantity = poolQuantityMap.get(pool.getId());
-            if (quantity > 0) {
-                quantityFound = true;
-            }
+        BindContext context = new BindContext(poolCurator, consumerCurator, i18n);
```

Comment:
The changes here are all happening within a transaction context. Unless \@Transactional is removed from the createEnittlements method I don't believe there will be a change in the lock time.